### ISSUE TITLE
`indent` rule false positive

### ISSUE THREAD

**Issue by feross** [eslint/eslint#5295]
Starting with eslint 2.1.0, I get incorrect `indent` behavior.

With this config:

```json
{
  "rules": {
    "indent": [2, 2]
  }
}
```

And this code:

```js
var info = {
  options: fn(
    true,
    {
      prop: true
    }
  )
}
```

I get this error:

```
/Users/feross/code/standard-stuff/test/t.js
  4:5  error  Expected indentation of 2 space characters but found 4  indent
  5:7  error  Expected indentation of 4 space characters but found 6  indent
  6:5  error  Expected indentation of 2 space characters but found 4  indent

✖ 3 problems (3 errors, 0 warnings)
```

In 2.0.0, this reported no errors. Starting in 2.1.0, it does.

This might be caused by @gyandeeps's fix for https://github.com/eslint/eslint/issues/5237 which was released in 2.1.0.

**Comment by gyandeeps** [comment]
@BYK Can you please evaluate this against your latest fix in #5237 ?

**Comment by abernix** [comment]
Same bug I believe... This is causing many of our builds to fail.  Here were my findings:

```javascript
/* eslint indent: [2,2] */
function SillyFunction(arg, arg2) {
  return { irrelevant: true };
}

// validates fine in 2.0 and 2.1
SillyFunction(
  {
    value: true,
  },
  {
    _id: true,
  }
);

var haveFun = function () {

  // No longer validates here in 2.1.0
  SillyFunction(
    {
      value: true,
    },
    {
      _id: true,
    }
  );

  // I guess this is what it wants?  This validates in 2.1.0.
  SillyFunction(
  {
    value: true,
  },
  {
    _id: true,
  }
  );

  return "fun";
};

// Guess not! Same code doesn't validate here, outside of the block.
SillyFunction(
{
  value: true,
},
{
  _id: true,
}
);

haveFun();
```

```bash
$ eslint --no-eslintrc untitled.js 

/private/tmp/untitled.js
  20:5  error  Expected indentation of 2 space characters but found 4  indent
  21:7  error  Expected indentation of 4 space characters but found 6  indent
  22:5  error  Expected indentation of 2 space characters but found 4  indent
  23:5  error  Expected indentation of 2 space characters but found 4  indent
  24:7  error  Expected indentation of 4 space characters but found 6  indent
  25:5  error  Expected indentation of 2 space characters but found 4  indent
  43:1  error  Expected indentation of 2 space characters but found 0  indent
  44:3  error  Expected indentation of 4 space characters but found 2  indent
  45:1  error  Expected indentation of 2 space characters but found 0  indent
  46:1  error  Expected indentation of 2 space characters but found 0  indent
  47:3  error  Expected indentation of 4 space characters but found 2  indent
  48:1  error  Expected indentation of 2 space characters but found 0  indent

✖ 12 problems (12 errors, 0 warnings)
```

**Comment by BYK** [comment]
@gyandeeps - definitely a fall out from that. Taking on, hope to get a fix out soon.

**Comment by WhyEvenTry** [comment]
Are you sure this was fixed? Code example below, the first version is fine, the second version causes indentation complaints. My setting is

    "indent": [1, 4, {"SwitchCase": 1}]

```
Promise.resolve()
.then(
    () => MessageBus.send('new-versioned-object', {
        obj: {},
        hash: 'foo'
    })
);

Promise.resolve()
.then(() => MessageBus.send('new-versioned-object', {
        obj: {},
        hash: 'foo'
    })
);
```

Screenshot (WebStorm: the yellow marks eslint errors):
![unbenannt-1](https://cloud.githubusercontent.com/assets/199179/13203374/e66e2b5e-d8b6-11e5-9232-34352d5a7c43.png)

eslint output:

```
  654:9   warning  Expected indentation of 4 space characters but found 8   indent
  655:9   warning  Expected indentation of 4 space characters but found 8   indent
  656:5   warning  Expected indentation of 0 space characters but found 4   indent
```
